ods_ci/build/install_idp_interop.sh

```diff
@@ -0,0 +1,149 @@
+if [[ -z "${CURRENT_IDP_LIST}" ]] || [[  "${CURRENT_IDP_LIST}" == "null" ]]; then
+  echo 'No oauth identityProvider exists. Initializing oauth .spec.identityProviders = []'
+  oc patch oauth cluster --type json -p '[{"op": "add", "path": "/spec/identityProviders", "value": []}]'
+fi
+
+# Patch in the LDAP identityProviders
+oc patch oauth cluster --type json -p '[{"op": "add", "path": "/spec/identityProviders/-", "value": '"$OAUTH_LDAP_JSON"'}]'
+
+$PROVIDER_PATH/ldap/ldap_installation.sh
+}
+
+add_groups_users() {
+# create groups
+oc adm groups new rhods-admins
+oc adm groups new rhods-users
+oc adm groups new rhods-noaccess
+oc adm groups new dedicated-admins
+# add users to groups
+function add_users_to_groups(){
+for i in {1..4}
+  do
+    oc adm groups add-users $1 $2$i 
+  done
+}
+add_users_to_groups rhods-admins htpasswd-admin
+add_users_to_groups rhods-users htpasswd-user
+add_users_to_groups rhods-noaccess htpasswd-noaccess
+add_users_to_groups rhods-admins ldap-admin
+add_users_to_groups dedicated-admins ldap-admin
+add_users_to_groups rhods-users ldap-user
+add_users_to_groups rhods-noaccess ldap-noaccess
+oc adm groups add-users dedicated-admins htpasswd-cluster-admin-user
+
+oc adm groups add-users rhods-admins kubeadmin
+oc adm policy add-cluster-role-to-group view rhods-admins
+oc adm policy add-cluster-role-to-group cluster-admin dedicated-admins
+
+oc describe oauth.config.openshift.io/cluster
+}
+
+function htpasswd_installation(){
+  chk_htpasswd=1
+
+  while read -r line; do
+
+    if [[ $line == *"htpasswd-cluster-admin"* ]]; then
+        echo -e "\033[0;33m Htpasswd Identity provider is installed. Skipping installation \033[0m"
+        chk_htpasswd=0
+        break
+    fi
+  done < <(oc get oauth -o yaml)
+
+  if [[ $chk_htpasswd == 1 ]]; then
+    install_htpasswd_identity_provider
+  fi
+}
+
+function ldap_installation(){
+  chk_ldap=1
+  while read -r line; do
+    if [[ $line == *"ldap-provider-qe"* ]]; then
+	    echo -e "\033[0;33m LDAP Identity provider is installed. Skipping installation \033[0m"
+	    chk_ldap=0
+        break
+    fi
+  done < <(oc get oauth -o yaml)
+  if [[ $chk_ldap == 1 ]]; then
+    install_ldap_identity_provider
+  fi
+}
+
+function createHtpasswdIDP(){
+    htpasswd -b -B -c $ARTIFACT_DIR/users.txt htpasswd-cluster-admin-user rhodsPW#123456
+    oc delete secret htpasswd-bind-password --ignore-not-found -n openshift-config
+    oc create secret generic htpasswd-bind-password --from-file=htpasswd=$ARTIFACT_DIR/users.txt -n openshift-config
+    oc delete identity htpasswd-cluster-admin:htpasswd-cluster-admin-user --ignore-not-found
+    oc patch oauth cluster --type json -p '[{op: add, path: /spec/identityProviders, value: []}]'
+    oc patch oauth cluster --type json -p '[{"op": "add", "path": "/spec/identityProviders/-", "value": {"name":"htpasswd-cluster-admin","mappingMethod":"claim","type":"HTPasswd","htpasswd":{"fileData":{"name":"htpasswd-bind-password"}}}}]'
```

Comment:
Maybe we should move oauth patches to a function?
also being used in lines 30, 34 and varies in `"comment": "notsecret"`

Reply:
We may address IDP user creation and security thing in a separate task. For now we prefer to use the existing code that are functioning now.